controllers/bookController.js

```diff
@@ -0,0 +1,101 @@
+import fs from "fs";
+import path from "path";
+import { ERRORS } from "../constants/constants";
+import { generateID } from "../utils/utils";
+
+// Get absolute path to the JSON file
+const booksFilePath = path.resolve("data/books.json");
+
+// Helper function to load data
+const loadBooks = () => {
+  try {
+    const data = fs.readFileSync(booksFilePath, "utf-8");
+    return JSON.parse(data);
+  } catch (error) {
+    console.error("Error loading books:", error);
+    return [];
+  }
+};
+
+// Save books
+const saveBooks = (books) => {
+  fs.writeFileSync(booksFilePath, JSON.stringify(books, null, 2));
+};
```

Comment:
Is this going to change the books.json file?

Reply:
Thank you, Matilda! ☺️🙏 
Yes, it writes to books.json and adds the new book at the end.